### PR TITLE
derive annotation from .db packages for affy

### PR DESCRIPTION
**Pull request by pinin4fjords**
These changes allow for production of a probe annotation table alongside affy array analysis. It's pretty easy to do with annotation packages.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/affy/justrma/main.nf [hunk 1]
```diff
@@ -12,9 +12,10 @@ process AFFY_JUSTRMA {
     tuple val(meta2), path(description)
 
     output:
-    tuple val(meta), path("*.rds"), emit: rds
-    tuple val(meta), path("*.tsv"), emit: tsv
-    path "versions.yml"           , emit: versions
+    tuple val(meta), path("*.rds")                           , emit: rds
+    tuple val(meta), path("matrix.tsv")                      , emit: expression
+    tuple val(meta), path("*.annotation.tsv"), optional: true, emit: annotation
+    path "versions.yml"                      , emit: versions
```
Comment (maxulysse) [inline modules/nf-core/affy/justrma/main.nf:18]:
```suggestion
    tuple val(meta), path("*.rds")           , emit: rds
    tuple val(meta), path("matrix.tsv")      , emit: expression
    tuple val(meta), path("*.annotation.tsv"), emit: annotation, optional: true
    path "versions.yml"                      , emit: versions
```

File: modules/nf-core/affy/justrma/meta.yml [hunk 2]
```diff
@@ -39,10 +39,14 @@ output:
...
+  - expression:
       type: file
       description: TSV-format intensity matrix
-      pattern: "*.tsv"
+      pattern: "matrix.tsv"
+  - annotation:
+      type: file
+      description: TSV-format annotation table
+      pattern: "*.annotation.tsv"
   - rdata:
```
Comment (maxulysse) [inline modules/nf-core/affy/justrma/meta.yml:50]:
```suggestion
  - rds:
```

**Review by maxulysse — COMMENTED** [review]
Minor comments

**Comment by pinin4fjords** [comment]
Thanks for comments @maxulysse - addressed

**Comment by pinin4fjords** [comment]
Thanks @maxulysse !

**Pull request merged**